tools/job-cache/job-cache.cpp

```diff
@@ -442,12 +641,41 @@ struct AddJobRequest {
       OutputFile output;
       output.source = output_file.second.get("src").value;
       output.path = output_file.second.get("path").value;
-      output.hash = output_file.second.get("hash").value;
+      output.hash = Hash256::from_hex(output_file.second.get("hash").value);
       outputs.emplace_back(std::move(output));
     }
   }
 };
 
+// TODO: Ideally we'd have someting like std::filepath
+template <class Iter>
+static std::string join(char sep, Iter begin, Iter end) {
```

Comment:
Can you document what these functions are supposed to do?

Looks like this one adds `sep` between each item in `Iter`?

Reply:
I think this is still open too